client/global.h

```diff
@@ -66,7 +66,12 @@ extern int g_disable_mining;
 //Test network and main network have different types of the block headers, so blocks from different networks are incompatible
 extern enum xdag_field_type g_block_header_type;
 
+#if defined (__MACOS__) || defined (__APPLE__)
+extern int is_pool(void);
+extern int is_wallet(void);
```

Comment:
What's the compilation error? it is strange to me, 
`inline int is_pool(void) { return g_xdag_type == XDAG_POOL; }
`
it is C standard... anything strange.

extern int is_pool(void); <-- it is not inlined anyway, it is only inlined in global.c file but not in other files